docs/design_docs/user_facing_api.md

```diff
@@ -0,0 +1,198 @@
+
+## Alternatives
+
+### Expose all host interfaces
+This approach would be valid for MVP.
+
+It would:
+  - read data from configMap
+  - if configMap provides empty configuration
+    - read all devices & bonds
+    - expose 1 resource per deviceOrBond with hard-coded values for:
+      - resource name: it would use the 'device' name
+      - mode: bridge
+      - capacity: 100
+
+This approach would fulfil 3/5 objectives, missing:
+- expose lower device as a logical resource
+- be able to use node selectors / node affinity / NIC selectors /
+resource name
+
+This approach is implemented in
+[PR #10](https://github.com/kubevirt/macvtap-cni/pull/10), and is considered
+good enough for the MVP.
+
+### Host device exposure using selectors
+This proposal complements the proposal where
+[all interfaces are exposed](#expose-all-host-interfaces). It would follow the
+same behavior of
+[k8s network policies](https://kubernetes.io/docs/concepts/services-networking/network-policies/#isolated-and-non-isolated-pods);
+by default all host's interfaces are exposed, but once an host is selected via
+node selector, a white-list mechanism is triggered.
+
+It would consist of 3 stages:
+  - tagging the hosts
+  - provisioning the macvtap policy
+  - provisioning the macvtap NAD
+
+#### Host tagging
+Host tagging could be based on:
+  - macvtapAware: bool (indicates if it is selectable via node selector)
+  - !node-role.kubernetes.io/master
+  - tag the nodes with the interface names you want it to expose (???)
+    - e.g. parent_iface.macvtap.network.kubevirt.io/eth0
+
+#### Macvtap policy
+The macvtap policy would look like this, should we choose a node-centric
+approach:
+```yaml
+apiVersion: ...
+kind: MacvtapNetworkNodePolicy
+metadata:
+  name: policy-1
+  namespace: macvtap-cni
+spec:
+  nicSelector:
+    - parentIface: bond1
+      mode: private
+      quantity: 25
+      resourceName: bondedIface
+    - parentIface: eth1
+      mode: vepa
+      resourceName: quickDataplane
+    - parentIface: eth2
+      resourceName: dataplane
+  nodeSelector:
+    matchLabels:
+      macvtapAware: true
+```
+For each node matching the `nodeSelector` knob, it would expose all interfaces
+on that host that are featured in the `nicSelector` list. Each resource would
+be advertised under the name specified in the `resourceName` attribute.
+
+Let's see an example, to better follow the proposal; there are 3 hosts,
+h1, h2, h3. The interfaces on each of them can be seen on the following yaml:
+
+```yaml
+h1:
+  - bond1
+  - eth0
+  - eth1
+h2:
+  - bond1
+  - ens3
+  - ens4
+  - ens5
+  - ens6
+h3:
+  - eth0
+  - eth1
+```
+Considering that nodes h1 and h2 **are tagged** with macvtapAware, and the
+cluster has been provisioned with the following policy:
+
+```yaml
+apiVersion: ...
+kind: MacvtapNetworkNodePolicy
+metadata:
+  name: policy-1
+  namespace: macvtap-cni
+spec:
+  nicSelector:
+    - parentIface: bond1
```

Comment:
When we are selecting nics, this is not really a parent yet. Can we change nicSelector to parentSelector and parentIface to just ifaceName?